### ISSUE TITLE
Bug with `terraform_required_providers` and data source checking

### ISSUE THREAD

**Issue by mgrecar** [terraform-linters/tflint#919]
I believe a change in v0.20.1 is now incorrectly flagging a violation with `terraform_required_providers`:
```
1 issue(s) found:

Warning: Missing version constraint for provider "terraform" in "required_providers" (terraform_required_providers)

  on data.tf line 1:
   1: data "terraform_remote_state" "vpc" {

Reference: https://github.com/terraform-linters/tflint/blob/v0.20.1/docs/rules/terraform_required_providers.md
```
However, that configuration has `required_providers` explicitly defined for all providers:
backend.tf
``` 
terraform {                                                                                                                                                                                                      
  required_version = "= 0.12.29"                                                                                                                                                                                 
                                                                                                                                                                                                                   
  required_providers {                                                                                                                                                                                           
    aws        = "= 3.3.0"                                                                                                                                                                                       
    kubernetes = "= 1.12.0"                                                                                                                                                                                      
    helm       = "= 1.2.4"                                                                                                                                                                                       
    kubectl    = "= 1.6.2"                                                                                                                                                                                       
  }
} 
```
Using TFLint v0.19.1, there's no issue.  This seems to be caused by https://github.com/terraform-linters/tflint/pull/889

The file flagged above looks like this:
data.tf
```
data "terraform_remote_state" "vpc" {                                                                                                                                                                            
  backend = "s3"                                                                                                                                                                                                 
                                                                                                                                                                                                                   
  config = {                                                                                                                                                                                                     
    bucket         = "my-terraform-state"                                                                                                                                                                       
    key            = "my-terraform-state/terraform.tfstate"                                                                                                                                                      
    role_arn       = "arn:aws:iam::xxxx:role/TerraformStateAccessRole"                                                                                                                                   
    external_id    = "Terraform"                                                                                                                                                                                 
    region         = "us-east-2"                                                                                                                                                                                 
    dynamodb_table = "terraform_state_locking_table"                                                                                                                                                             
  }                                                                                                                                                                                                              
}  
```
I could be misunderstanding something, but I think everything is appropriately locked down in my scenario.

## Version
TFLint v0.20.1
Terraform v0.12.29

**Comment by bendrucker** [comment]
Double checking whether it's still embedded in core or on the registry now. There is definitely a provider called "terraform" that implements the remote state data source, but last I looked it was embedded in the Terraform CLI. If that's still the case the fix will be hardcoding/ignoring this provider.

**Comment by bendrucker** [comment]
Seems to be the reverse—it was briefly part of the registry but is embedded in core and the registry version is no longer updated:

https://github.com/hashicorp/terraform-provider-terraform
